core/src/main/java/kafka/server/ReplicaFetcherTierStateMachine.java

```diff
@@ -216,6 +217,10 @@ private Long buildRemoteLogAuxState(TopicPartition topicPartition,
                 }
             }
 
+            if (!rlm.isInitialized(topicPartition)) {
```

Comment:
Please add a comment on why this check is helpful as you mentioned it in the PR description.